### PR TITLE
[Header]: First drop Header component

### PR DESCRIPTION
**Pull request by davidicus**
**Summary**

- The first section of the UI-Shell. Top header that allows for dropdown menus triggered by the action buttons. 

**Change List (commits, features, bugs, etc)**

- added Header.jsx
- added HeaderMenu.jsx

**Related Issues**

- First part of https://github.com/IBM/carbon-addons-iot-react/issues/27

**Acceptance Test (how to verify the PR)**

- Able to trigger both actions and dropdown when clicking on action buttons.

### CODE REVIEW

**Comment by davidicus** [comment]
The UI-shell requires a flag to be set before carbon styles are applied. This flag is set in storybook but not sure how we need to document this.

File: package.json [hunk 1]
```diff
@@ -29,10 +29,12 @@
     }
   },
   "dependencies": {
+    "@carbon/icons-react": "^0.0.1-alpha.12",
```
Comment (tay1orjones) [inline package.json]: We already have this as a peer dep, is there a reason to duplicate into dep?

**Comment by scottdickerson** [comment]
We should put it in devdeps for storybook

Sent from my iPhone

On Mar 6, 2019, at 3:45 PM, Taylor Jones <notifications@github.com<mailto:notifications@github.com>> wrote:


@tay1orjones commented on this pull request.

________________________________

In package.json<https://github.com/IBM/carbon-addons-iot-react/pull/63#discussion_r263149452>:

> @@ -29,10 +29,12 @@
     }
   },
   "dependencies": {
+    "@carbon/icons-react": "^0.0.1-alpha.12",


We already have this as a peer dep, is there a reason to duplicate into dep?

**Comment by davidicus** [comment]
I restarted the last previously passing commit and it is failing now. I think there may be an issue with Travis.

File: package.json (hunk 1 above)
Comment (davidicus) [reply package.json]: I removed and it worked. I missed the peerdep/

**Pull request merged**